lib/public/views/QcFlags/Overview/QcFlagsOverviewModel.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * @license
+ * Copyright CERN and copyright holders of ALICE O2. This software is
+ * distributed under the terms of the GNU General Public License v3 (GPL
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+import { buildUrl } from '../../../utilities/fetch/buildUrl.js';
+import { OverviewPageModel } from '../../../models/OverviewModel.js';
+import { RemoteData } from '/js/src/index.js';
+import { getRemoteData } from '../../../utilities/fetch/getRemoteData.js';
+import { ObservableData } from '../../../utilities/ObservableData.js';
+import { detectorsProvider } from '../../../services/detectors/detectorsProvider.js';
+
+/**
+ * Quality Control Flags overview model
+ *
+ * @implements {OverviewModel}
+ * @abstract
+ */
+export class QcFlagsOverviewModel extends OverviewPageModel {
+    /**
+     * The constructor of the Overview model object
+     */
+    constructor() {
+        super();
+        this._run$ = new ObservableData(RemoteData.notAsked());
+        this._run$.bubbleTo(this);
+        this._dataProduction$ = new ObservableData(RemoteData.notAsked());
+        this._dataProduction$.bubbleTo(this);
+        this._dplDetector$ = ObservableData.builder()
+            .initialValue(RemoteData.notAsked())
+            .source(detectorsProvider.dpl$)
+            .apply((remoteData) => remoteData.apply({
+                Success: (detectors) => detectors.find(({ id }) => id === this._dplDetectorId),
+            }))
+            .build();
+        this._dplDetector$.bubbleTo(this);
+    }
+
+    // eslint-disable-next-line valid-jsdoc
+    /**
+     * @inheritdoc
+     */
+    getRootEndpoint() {
+        const params = {
+            filter: {
+                runNumbers: [this._runNumber],
+                dplDetectorIds: [this._dplDetectorId],
+            },
+        };
+        return buildUrl('/api/qcFlags', params);
+    }
+
+    // eslint-disable-next-line valid-jsdoc
+    /**
+     * @inheritdoc
+     */
+    async load() {
+        detectorsProvider.dpl$.notify();
+        this._fetchRun();
+        this._fetchDataProduction();
```

Comment:
Are you sure it makes sense to factorize this?
Why not simply overriding the load function in both child and call custom method from here?
Because what you load is not really the same thing I think

Reply:
applied